src/Jobs/Recruit.Vacancies.Jobs/Program.cs

```diff
@@ -34,152 +21,63 @@ static Program()
-            jobConfiguration.JobActivator = new CustomJobActivator(serviceProvider);
-            jobConfiguration.UseTimers();
-            jobConfiguration.LoggerFactory = loggerFactory;
-            
-            if (IsDevelopment)
+            var builder = new HostBuilder()
+                .UseEnvironment(EnvironmentName)
+                .ConfigureWebJobs(b =>
+                {
+                    b.AddAzureStorageCoreServices()
+                    .AddAzureStorage()
+                    .AddTimers();
+                })
+                .ConfigureAppConfiguration(b =>
+                {
+                    b.AddJsonFile("appSettings.json", optional: false)
+                    .AddJsonFile($"appSettings.{EnvironmentName}.json", true)
+                    .AddEnvironmentVariables();
+
+                    if (IsDevelopment)
+                    {
+                        b.AddUserSecrets<Program>();
+                    }
+                })
+                .ConfigureLogging((context, b) =>
+                {
+                    b.SetMinimumLevel(LogLevel.Trace);
+                    b.AddDebug();
+                    b.AddConsole();
+                    b.AddNLog();
+
+                    // If this key exists in any config, use it to enable App Insights
+                    string appInsightsKey = context.Configuration["APPINSIGHTS_INSTRUMENTATIONKEY"];
+                    if (!string.IsNullOrEmpty(appInsightsKey))
+                    {
+                        b.AddApplicationInsights(o => o.InstrumentationKey = appInsightsKey);
+                    }
+                })
+                .ConfigureServices((context, services) =>
+                {
+                    services.AddSingleton<IQueueProcessorFactory, CustomQueueProcessorFactory>();
+                    services.Configure<QueuesOptions>(options =>
+                    {
+                        //maximum number of queue messages that are picked up simultaneously to be executed in parallel (default is 16)
+                        options.BatchSize = 1;
+                        //Maximum number of retries before a queue message is sent to a poison queue (default is 5)
+                        options.MaxDequeueCount = 5;
+                        //maximum wait time before polling again when a queue is empty (default is 1 minute).
+                        options.MaxPollingInterval = TimeSpan.FromSeconds(10);
+                    });
+
+                    services.ConfigureJobServices(context.Configuration);
+                })
+                .UseConsoleLifetime();
+
+            var host = builder.Build();
+            using (host)
             {
-                jobConfiguration.DashboardConnectionString = null; // Reduces errors in output.
-                jobConfiguration.UseDevelopmentSettings();
-                jobConfiguration.Tracing.ConsoleLevel = TraceLevel.Off;
+                await host.RunAsync();
```

Comment:
Just wondering if there was an uncaught exception triggering the job or some of the jobs that don't have try/catches would we not want the job host to catch, log and and maybe re-throw. Thinking of the cases of the `This job has met a horrible end` scenarios.

Reply:
From what I can see any unhandled exceptions from the functions don't kill the host and logs out that that particular function has failed.